chef/cookbooks/glance/recipes/ceph.rb

```diff
@@ -24,17 +27,19 @@
 else
   # If external Ceph cluster will be used,
   # we need install ceph client packages
-  if node[:platform] == "suse" && File.exists?("/etc/ceph/ceph.client.admin.keyring")
-    package "ceph-common"
-    package "python-ceph"
-  else 
+  if File.exists?(ceph_conf) && File.exists?(admin_keyring)
+    if node[:platform] == "suse"
+      package "ceph-common"
+      package "python-ceph"
+    end
+  else
     return
   end
 end
 
 # If ceph.conf and admin keyring will be available 
 # we have to check ceph cluster status
-check_ceph = Mixlib::ShellOut.new("ceph -s | grep -q -e 'HEALTH_[OK|WARN]'")
+check_ceph = Mixlib::ShellOut.new("ceph -k #{admin_keyring} -c #{ceph_conf} -s | grep -q -e 'HEALTH_[OK|WARN]'")
```

Comment:
I guess technially we'd need to shell escape ```admin_keyring``` and ```ceph_conf``` here as well? At least if we expect file names with spaces and other funny characters to work here.